includes/class-sensei-temporary-user.php

```diff
@@ -35,6 +35,8 @@ public static function init() {
 		add_filter( 'sensei_learners_query', [ static::class, 'filter_learners_query' ] );
 		add_filter( 'sensei_count_statuses_args', [ static::class, 'filter_count_statuses' ] );
 		add_filter( 'sensei_check_for_activity', [ static::class, 'filter_sensei_activity' ], 10, 2 );
+		add_filter( 'pre_wp_mail', [ Sensei_Guest_User::class, 'skip_wp_mail' ], 10, 2 );
+		add_filter( 'pre_wp_mail', [ Sensei_Preview_User::class, 'skip_wp_mail' ], 10, 2 );
```

Comment:
I'm not a huge fan of this change: basically, I'm adding the filters here for each class. 

I thought about maybe creating a method in each class just to run that `add_filter` and call it here, but I think the result would be...quite similar, so that's why I left it this way.

This is also the main change that fixes the issue, so..yeah, there's that.